src/o365/spo/commands/field/field-remove.ts

```diff
@@ -81,26 +78,71 @@ class SpoFieldRemoveCommand extends SpoCommand {
         json: true
       };
 
-      request
-        .post(requestOptions)
+      return request.post(requestOptions);
+    }
+    
+    const prepareRemoval = (): void => {
+      let listRestUrl: string = '';
+
+      if (args.options.listId) {
+        listRestUrl = `lists(guid'${encodeURIComponent(args.options.listId)}')/`;
+      }
+      else if (args.options.listTitle) {
+        listRestUrl = `lists/getByTitle('${encodeURIComponent(args.options.listTitle as string)}')/`;
+      }
+      else if (args.options.listUrl) {
+        const listServerRelativeUrl: string = Utils.getServerRelativePath(args.options.webUrl, args.options.listUrl);
+        listRestUrl = `GetList('${encodeURIComponent(listServerRelativeUrl)}')/`;
+      }
+
+      if (args.options.group) {
+        if (this.verbose) {
+          cmd.log(`Retrieving fields assigned to group ${args.options.group}...`);
+        }
+        const requestOptions: any = {
+          url: `${args.options.webUrl}/_api/web/${listRestUrl}fields`,
+          headers: {
+            accept: 'application/json;odata=nometadata'
+          },
+          json: true
+        };
+  
+        request
+          .get(requestOptions)
+          .then((res: any): void => {
+            let filteredResults = res.value.filter((field: { Id: string | undefined, Group: string | undefined; }) => field.Group === args.options.group);
+            if (this.verbose) {
+              cmd.log(`${filteredResults.length} matches found...`);
+            }
+
+            var promises = [];
+            for (let index = 0; index < filteredResults.length; index++) {
+              promises.push(removeField(listRestUrl, filteredResults[index].Id, undefined));
+            }
+            
+            Promise.all(promises).then(() => {
+              cb();
+            })
+            .catch((err) => {
+              this.handleRejectedODataJsonPromise(err, cmd, cb);
+            });
+          }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+      }
+      else {
```

Comment:
We could simplify this piece of code by always calling `Promise.all`. In case only one field was specified, we'd push to the array of promises only one call to `removeField`. That way we can avoid duplication of calling `removeField`, handling errors and completing command execution.

Reply:
I agree that we should simplify but I am not sure how to handle this properly. I am struggling to solve it without an await function.

When handling multiple fields I have to wait until all fields are returned until I could push the items into the remove field promises. For an individual field, I do not have to wait for this operation to run.

![image](https://user-images.githubusercontent.com/18241305/84585750-2f560680-ae13-11ea-83d6-1caf7ed8b1e5.png)

When taking the Promise.all out I am no more waiting for all fields to return when looking for multiple fields.

Reply:
You're right. Sorry, I missed that. Let's keep it as-is. Sorry for the trouble.

Reply:
That said, I'll continue the review